api/v1/storagecluster_types.go

```diff
@@ -237,6 +237,10 @@ type ManageCephFilesystems struct {
 	// +kubebuilder:validation:MaxLength=253
 	// +kubebuilder:validation:Pattern=^[a-z0-9]([-a-z0-9]*[a-z0-9])?(\\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*$
 	StorageClassName string `json:"storageClassName,omitempty"`
+	// DataPool specifies the pool specification for the default cephfs data pool
+	DataPool rookCephv1.NamedPoolSpec `json:"dataPool,omitempty"`
+	// AdditionalDataPools specifies the pool specification for the additional cephfs data pools
+	AdditionalDataPools []rookCephv1.NamedPoolSpec `json:"additionalDataPools,omitempty"`
```

Comment:
why we need two fields for the DataPool? cant we just have one which accepts an array of dataPools?

Reply:
We shouldn't go changing the existing fields, so `DataPool` would still have to remain. I believe there was an idea to have just a `DataPools` field and to "deprecate" `DataPool`, but... that's iffy, and introduces support and QE challenges.

Reply:
In the [design doc](https://docs.google.com/document/d/1N0lqswqU5iAqnrhThPTAUHtwupzxGY_L5dqjYM5ih4I/edit#heading=h.edvemstuogz2) some time ago it had been determined that it will be most clear to have the dataPool field that corresponds to the default pool that is created with the filesystem. Consider two scenarios:
1. The admin wants to add a replica 2 pool without changing the default replica 3 pool. They would add the pool spec to the `additionalDataPools`, but not define the default `dataPool`.
2. The admin wants to customize the pool settings of the default pool (and even change it to replica 2). They would add/modify the `dataPool` field.

The first point is the basis for why to not have the pools all in a single slice since adding the replica 2 pool in addition to the default pool would also confusingly require adding the spec for the default replica 3 pool.

Reply:
@jarrpa The default data pool is always there regardless of the existence of the DataPool configuration field. The section itself does not indicate the existence of that default data pool, it just gives a way to attach configuration to it.

On the other hand, the AdditionalDataPools define the existence of these additional data pools (together with their configurations). So with that in mind, it make sense to have these 2 configuration separate